domain/secretbackend/state/state.go

```diff
@@ -46,18 +46,42 @@ func NewState(factory coredatabase.TxnRunnerFactory, logger Logger) *State {
 // GetModel is responsible for returning the model for the provided uuid. If no
 // model is found the given uuid then an error of
 // [github.com/juju/juju/domain/model/errors.NotFound] is returned.
-func (s *State) GetModel(ctx context.Context, uuid coremodel.UUID) (coremodel.Model, error) {
+func (s *State) GetModel(ctx context.Context, uuid coremodel.UUID) (*secretbackend.Model, error) {
 	db, err := s.DB()
 	if err != nil {
-		return coremodel.Model{}, errors.Trace(err)
+		return nil, errors.Trace(err)
 	}
 
-	var m coremodel.Model
-	return m, db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
-		var err error
-		m, err = modelstate.Get(ctx, tx, uuid)
-		return err
+	stmt, err := sqlair.Prepare(`
+SELECT (
+	uuid, name, model_type_type, secret_backend_uuid
+) AS (&Model.*)
+FROM v_model
+WHERE uuid = $M.uuid`, sqlair.M{}, Model{})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	var m Model
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt, sqlair.M{"uuid": uuid}).Get(&m)
+		if errors.Is(err, sql.ErrNoRows) {
+			return fmt.Errorf("%w: %q", modelerrors.NotFound, uuid)
+		}
+		return errors.Trace(err)
 	})
+	if err != nil {
+		return nil, errors.Trace(domain.CoerceError(err))
+	}
+	if !m.Type.IsValid() {
+		// This should never happen.
+		return nil, fmt.Errorf("invalid model type for model %q", m.Name)
+	}
+	return &secretbackend.Model{
```

Comment:
Then with these as values, you can use the `copy` method to make this a 2-liner.